scripts/extensions/ai-widget/src/ai-assistant.tsx

```diff
@@ -1,74 +1,185 @@
+            };
+        } else {
+            this.state = {
+                activeSection: null,
+            };
+        }
+
         this.setError = this.setError.bind(this);
         this.generateHeadlines = this.generateHeadlines.bind(this);
         this.generateSummary = this.generateSummary.bind(this);
+        this.generateTranslations = this.generateTranslations.bind(this);
     }
 
     setError() {
-        this.setState({
-            error: true,
-        });
+        const state = this.state;
+
+        if (state.activeSection != null) {
+            this.setState({...state, error: true, loading: false});
+        }
     }
 
     generateHeadlines() {
-        configuration.generateHeadlines?.(this.props.article, superdesk)
-            .then((res) => {
-                this.setState({
-                    loadingHeadlines: false,
-                    headlines: res,
+        const state = this.state;
+
+        if (state.activeSection === 'headlines') {
+            configuration.generateHeadlines?.(this.props.article, superdesk)
+                .then((res) => {
+                    this.setState({
+                        ...state,
+                        loading: false,
+                        headlines: res,
+                    });
+                }).catch(() => {
+                    this.setError();
                 });
-            }).catch(() => {
-                this.setError();
-            });
+        }
+    }
+
+    generateTranslations() {
+        const state = this.state;
+
+        if (state.activeSection === 'translations') {
+            configuration.translations?.generateTranslations?.(this.props.article, state.activeLanguageId, superdesk)
+                .then((res) => {
+                    this.setState({
+                        ...state,
+                        loading: false,
+                        translations: res,
+                    });
+                }).catch(() => {
+                    this.setError();
+                });
+        }
     }
 
     generateSummary() {
-        configuration.generateSummary?.(this.props.article, superdesk)
-            .then((res) => {
-                this.setState({
-                    loadingSummary: false,
-                    summary: res,
+        const state = this.state;
+
+        if (state.activeSection === 'summary') {
+            configuration.generateSummary?.(this.props.article, superdesk)
+                .then((res) => {
+                    this.setState({
+                        ...state,
+                        loading: false,
+                        summary: res,
+                    });
+                }).catch(() => {
+                    this.setError();
                 });
-            }).catch(() => {
-                this.setError();
-            });
+        }
+    }
+
+    componentDidUpdate(_prevProps: Readonly<IArticleSideWidgetComponentType>, prevState: Readonly<IState>): void {
```

Comment:
you have already `setSection` why not handle storing/retrieving inactive tab states there?